client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,257 @@
+
+	getCurrentVerificationCodes() {
+		return {
+			google: this.refs.verification_code_google.props.value,
+			bing: this.refs.verification_code_bing.props.value,
+			pinterest: this.refs.verification_code_pinterest.props.value,
+			yandex: this.refs.verification_code_yandex.props.value
+		};
+	},
+
+	submitSeoForm( event ) {
+		var site = this.props.site;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		this.setState( { submittingForm: true } );
+
+		const updatedOptions = {
+			seo_meta_description: this.state.seo_meta_description,
+			verification_services_codes: this.state.verification_services_codes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { submittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { submittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+	},
+
+	render() {
+		const { site } = this.props;
+		const { submittingForm, seo_meta_description, verification_services_codes } = this.state;
+
+		const blog_public = get( site, 'settings.blog_public', 1 );
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
+
+		const fetchingSettings = get( site, 'fetchingSettings', false );
+		const isDisabled = isSitePrivate || fetchingSettings || submittingForm;
+		const isMetaError = seo_meta_description && seo_meta_description.length > 160;
+		const sitemapUrl = `https://${ site.slug }/sitemap.xml`;
+		const generalSettingsUrl = `/settings/general/${ site.slug }`;
+
+		// The API returns 'false' for an empty array value, so we force it to an empty string if needed
+		const googleCode = get( verification_services_codes, 'google' ) || '';
+		const bingCode = get( verification_services_codes, 'bing' ) || '';
+		const pinterestCode = get( verification_services_codes, 'pinterest' ) || '';
+		const yandexCode = get( verification_services_codes, 'yandex' ) || '';
+
+		return (
+			<div className={ this.state.fetchingSettings ? 'is-loading' : '' }>
+				{ isSitePrivate &&
+					<Notice status="is-warning" showDismiss={ false } text={ this.translate( 'SEO settings are disabled because the site visibility is not set to Public.' ) }>
+						<NoticeAction href={ generalSettingsUrl }>{ this.translate( 'View Settings' ) }</NoticeAction>
+					</Notice>
+				}
+				<SectionHeader label={ this.translate( 'Search Engine Optimization' ) }>
+					<Button
+						compact={ true }
+						onClick={ this.submitSeoForm }
+						primary={ true }
+
+						type="submit"
+						disabled={ isDisabled || this.state.submittingForm }>
```

Comment:
I'll only mention this once, since this PR already has many comments. I'm personally a fan of bringing the `>` down to the next line to help separate the indentation, though I'm not sure off-hand if we have a Calypso policy on this one way or the other.

```js
<Button
	compact={ true }
	onClick={ this.submitSeoForm }
>
	{ this.state.submittingForm
		? this.translate( 'Saving…' )
		: this.translate( 'Save Settings' ) }
</Button>
```

Okay, I'm not consistent here either because of how I ended the `}`, but I like both options with the curly brackets.